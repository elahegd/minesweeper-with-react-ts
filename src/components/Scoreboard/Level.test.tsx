import React from 'react';
import { render, screen, } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import {Level} from "./Level";

describe("Render Level component with props", () => {
    it("Level renders correctly", () => {
        const { asFragment } = render(<Level onChange={jest.fn()} value={'intermediat'}>{['beginner', 'intermediat', 'expert']}</Level>);

        expect(asFragment()).toMatchSnapshot();;
    });
});

describe("Change handler behaviour", () => {
    it("Change from intermediate to expert", () => {
        const onChange = jest.fn();

        render(<Level onChange={onChange} value={'intermediat'}>{['beginner', 'intermediat', 'expert']}</Level>);

        userEvent.selectOptions(screen.getByRole('combobox'), 'expert');

        expect(screen.getByRole('option', {name : 'expert'})).toBeEnabled();
        expect(onChange).toHaveBeenCalled();
    })
})