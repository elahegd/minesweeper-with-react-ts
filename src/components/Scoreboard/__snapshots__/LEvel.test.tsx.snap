// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Level component with props Level renders correctly 1`] = `
<DocumentFragment>
  <select
    class="css-1lpmfg0"
  >
    <option
      class="css-zcmdxr"
      value="beginner"
    >
      beginner
    </option>
    <option
      class="css-zcmdxr"
      value="intermediat"
    >
      intermediat
    </option>
    <option
      class="css-zcmdxr"
      value="expert"
    >
      expert
    </option>
  </select>
</DocumentFragment>
`;
